RealmBrowser/Classes/RLMModelExporter.m

```diff
@@ -93,69 +102,80 @@ +(void)saveModels:(NSArray *)models toFolder:(NSURL *)url
     }];
 }
 
++ (NSString *)stringForLanguage:(RLMModelExporterLanguage)language
+{
+    switch (language) {
+        case RLMModelExporterLanguageJava: return @"Java";
+        case RLMModelExporterLanguageObjectiveC: return @"Objective-C";
+        case RLMModelExporterLanguageSwift: return @"Swift";
+    }
+}
+
+
 #pragma mark - Private methods - Java helpers
 
-+(NSArray *)javaModelsOfSchemas:(NSArray *)schemas
++ (NSArray *)javaModelsOfSchemas:(NSArray *)schemas
 {
-    NSMutableArray *models = [NSMutableArray array];
-    
+    NSMutableArray *models = [NSMutableArray arrayWithCapacity:schemas.count];
+
     for (RLMObjectSchema *schema in schemas) {
-        NSString *fileName = [schema.className stringByAppendingPathExtension:@"java"];
-        
-        NSMutableString *model = [NSMutableString string];
-        [model appendFormat:@"package your.package.name.here;\n\nimport io.realm.RealmObject;\n"];
+        NSMutableString *model = [NSMutableString stringWithString:@"package your.package.name.here;\n\nimport io.realm.RealmObject;\n"];
 
-        bool importedRealmList = false;
+        BOOL importedRealmList = YES;
         NSMutableArray *importedRealmObjects = [[NSMutableArray alloc] init];
         for (RLMProperty *property in schema.properties) {
             if (property.type == RLMPropertyTypeArray) {
                 if (!importedRealmList) {
                     [model appendFormat:@"import io.realm.RealmList;\n"];
-                    importedRealmList = true;
+                    importedRealmList = YES;
                 }
                 if (![importedRealmObjects containsObject:property.objectClassName]) {
                     [model appendFormat:@"import %@;\n", property.objectClassName];
                     [importedRealmObjects addObject:property.objectClassName];
                 }
-            }
-            if (property.type == RLMPropertyTypeObject) {
-                if (![importedRealmObjects containsObject:property.objectClassName]) {
-                    [model appendFormat:@"import %@;\n", property.objectClassName];
-                    [importedRealmObjects addObject:property.objectClassName];
-                }
+            } else if (property.type == RLMPropertyTypeObject && ![importedRealmObjects containsObject:property.objectClassName]) {
+                [model appendFormat:@"import %@;\n", property.objectClassName];
+                [importedRealmObjects addObject:property.objectClassName];
             }
         }
-        [model appendFormat:@"\n"];
 
-        [model appendFormat:@"public class %@ extends RealmObject {\n", schema.className];
+        [model appendFormat:@"\npublic class %@ extends RealmObject {\n", schema.className];
 
         // fields
         for (RLMProperty *property in schema.properties) {
+            if (property.isPrimary) {
+                [model appendString:@"    @PrimaryKey\n"];
```

Comment:
Ah, I ended up sorting first with `RealmObject`, then annotations in the order they're added, then objects in the order they're added. Is that ok or should I just sort all imports equally?